src/vs/workbench/common/editor/textEditorModel.ts

```diff
@@ -114,6 +117,10 @@ export class BaseTextEditorModel extends EditorModel implements ITextEditorModel
 		const lang = await this.languageDetectionService.detectLanguage(this.textEditorModelHandle);
 		if (lang && !this.isDisposed()) {
 			this.setModeInternal(lang);
+			const languageName = this.modeService.getLanguageName(lang);
+			if (languageName) {
+				this.ariaAlertService?.alert(localize('languageAutoDetected', "Language {0} was automatically detected and set as the language mode.", languageName));
```

Comment:
not sure if the `?` is needed because if this model is used in non-browser targets... I don't really know what will happen.

Reply:
That should not be needed, there would be errors if someone was trying to instantiate this model without an alert service implementation.